src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerContext.cs

```diff
@@ -94,5 +98,18 @@ protected JsonSerializerContext(JsonSerializerOptions? options)
         /// <param name="type">The type to fetch metadata about.</param>
         /// <returns>The metadata for the specified type, or <see langword="null" /> if the context has no metadata for the type.</returns>
         public abstract JsonTypeInfo? GetTypeInfo(Type type);
+
+        JsonTypeInfo? IJsonTypeInfoResolver.GetTypeInfo(Type type, JsonSerializerOptions options)
+        {
+            if (_options != options)
+            {
+                Debug.Assert(_options != null, "_options are null");
+                ThrowHelper.ThrowInvalidOperationException_SerializerContextOptionsImmutable();
```

Comment:
I prefer to keep any behavioral changes separate to this PR. Also JsonSerializerContext is generated for specific options and they may end up being incompatible. Let's define the behavior when tests are added

Reply:
This is _introducing_ new behavior, so I'm not sure if such a distinction applies. I would strongly encourage you to introduce tests in tandem with new behavior instead of doing it in separate PRs.